src/main/java/org/openrewrite/java/migrate/jakarta/UpdateManagedBeanToNamed.java

```diff
@@ -0,0 +1,85 @@
+
+    @Override
+    public String getDisplayName() {
+        return "Update Faces `@ManagedBean` to use CDI `@Named`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Faces ManagedBean was deprecated in JSF 2.3 (EE8) and removed in Jakarta Faces 4.0 (EE10). " +
+               "Replace `@ManagedBean` with `@Named` for CDI-based bean management.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            private final AnnotationMatcher MANAGED_BEAN_MATCHER_JAVAX = new AnnotationMatcher("javax.faces.bean.ManagedBean");
+            private final AnnotationMatcher MANAGED_BEAN_MATCHER_JAKARTA = new AnnotationMatcher("jakarta.faces.bean.ManagedBean");
+
+            @Override
+            public J.Annotation visitAnnotation(J.Annotation annotation, ExecutionContext ctx) {
+                if (MANAGED_BEAN_MATCHER_JAVAX.matches(annotation) || MANAGED_BEAN_MATCHER_JAKARTA.matches(annotation)) {
+                    // Get the name from the @ManagedBean annotation
+                    String beanName;
+                    beanName = annotation.getArguments() == null ? null :
+                            annotation.getArguments().stream()
+                                    .filter(J.Assignment.class::isInstance)
+                                    .map(J.Assignment.class::cast)
+                                    .filter(arg -> arg.getVariable().toString().equals("name"))
+                                    .findFirst()
+                                    .map(arg -> arg.getAssignment().toString())
+                                    .orElse(null);
+                    // Replace the @ManagedBean annotation with @Named
+                    if (beanName != null) {
+                        annotation = JavaTemplate.builder("@Named(\"#{}\")")
+                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "jakarta.inject-api-2.0.1"))
+                                .imports("jakarta.inject.Named")
+                                .build()
+                                .apply(getCursor(), annotation.getCoordinates().replace(), beanName);
+                    } else {
+                        annotation = JavaTemplate.builder("@Named")
+                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "jakarta.inject-api-2.0.1"))
+                                .imports("jakarta.inject.Named")
+                                .build()
+                                .apply(getCursor(), annotation.getCoordinates().replace());
+                    }
+                    maybeAddImport("jakarta.inject.Named");
+                    maybeRemoveImport("javax.faces.bean.ManagedBean");
+                    maybeRemoveImport("jakarta.faces.bean.ManagedBean");
+                }
+                return annotation;
```

Comment:
If there's any chance of these appearing in nested annotations, then it might make sense to call `super.visitAnnotation` here.

Reply:
I don't think there's any chance of nested annotations here.